src/ess/powder/correction.py

```diff
@@ -257,6 +257,86 @@ def _shallow_copy(da: sc.DataArray) -> sc.DataArray:
     return out
 
 
+def compute_pdf_from_structure_factor(
+    s: sc.DataArray,
+    *,
+    rho: sc.Variable,
+    r: sc.Variable,
+    self_scattering: sc.Variable,
+    use_filter: bool,
+) -> sc.DataArray:
+    '''
+    Computes the pair-distribution-function :math:`D(r)` as defined in
+    `Review: Pair distribution functions from neutron total scattering for the study of local structure in disordered materials <https://www.sciencedirect.com/science/article/pii/S2773183922000374>`_
+    from the overall scattering function :math:`S(Q)`.
+
+    The inputs to the algorithm are:
+
+    * A histogram representing :math:`S(Q)` with :math:`N` bins on a bin-edge grid with :math:`N+1` edges :math:`Q_j` for :math:`j=0\\ldots N`.
+    * The density :math:`\\rho`, the total number of atoms per volume.
+    * The structure independent coherent scattering amplitude :math:`S_{self} = \\sum_m c_m b_m^2` where the sum is over the atomic species in the sample, :math:`c_m` is the fraction of atoms that belong to species :math:`m` and :math:`b_m` is the size of scattering from an atom of that species.
+    * The bin-edge grid over :math:`r` the output histogram representing :math:`D(r)` will be computed on.
+
+    In each output bin, the output is computed as:
+
+    .. math::
+        D_{i+\\frac{1}{2}} = \\frac{2}{\\pi(r_{i+1}-r_i)} \\int_{r_i}^{r_{i+1}} \\int_{0}^\\infty (S(Q) - S_{self}) Q sin(Q r) dQ \\ dr  \\\\
+        \\approx \\frac{2}{\\pi(r_{i+1}-r_i)} \\sum_{j=0}^{N-1} (S(Q)_{j+\\frac{1}{2}} - S_{self}) (cos(\\bar{Q}_{j+\\frac{1}{2}} r_{i})-cos(\\bar{Q}_{j+\\frac{1}{2}} r_{i+1})) \\Delta Q_{j+\\frac{1}{2}}
+
+    Note that in the above expression the subscript :math:`_{j+\\frac{1}{2}}` is used to denote
+    quantities belonging to the :math:`j`:th bin of a histogram, :math:`\\bar{Q}_{j+\\frac{1}{2}} = \\frac{Q_j + Q_{j+1}}{2}` and :math:`\\Delta Q_{j+\\frac{1}{2}} = Q_{j+1} - Q_{j}`.
```

Comment:
```suggestion
    quantities belonging to the :math:`j^\\text{th}` bin of a histogram, :math:`\\bar{Q}_{j+\\frac{1}{2}} = \\frac{Q_j + Q_{j+1}}{2}` and :math:`\\Delta Q_{j+\\frac{1}{2}} = Q_{j+1} - Q_{j}`.
```
Because `j:th` is Swedish spelling, as far as I can tell.

Reply:
Good to know, thanks.